dm/dm/master/server.go

```diff
@@ -861,22 +864,30 @@ func (s *Server) UnlockDDLLock(ctx context.Context, req *pb.UnlockDDLLockRequest
 		return resp, nil
 	}
 	subtasks := s.scheduler.GetSubTaskCfgsByTask(task)
+	unlockType := config.ShardPessimistic
 	if len(subtasks) > 0 {
 		// subtasks should have same ShardMode
 		for _, subtask := range subtasks {
 			if subtask.ShardMode == config.ShardOptimistic {
-				resp.Msg = "`unlock-ddl-lock` is only supported in pessimistic shard mode currently"
+				unlockType = config.ShardOptimistic
```

Comment:
```suggestion
				unlockType = sbutasks[0]. ShardMode
```

all subtasks of one task have same shardMode, so we can move this check from for-loop ( move this line to L871)

Reply:
addressed in https://github.com/pingcap/tiflow/pull/4087/commits/7b561d67f67afae4258ab994d936eb731f9c5223